### PR TITLE
add a file fetcher & proxy cmd args

### PR DESCRIPTION
**Pull request by wrfly**

### CODE REVIEW

**Comment by wrfly** [comment]
ping @jpillora

**Comment by jpillora** [comment]
Thanks! I've been meaning to add a file fetcher. Will review soon

**Comment by wrfly** [comment]
Em..... A long soon =.=

ping @jpillora

**Comment by jpillora** [comment]
Oh crap, will take a look now :)

File: fetcher/fetcher_file.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+	"time"
+)
+
+// FILE is used to check new version
+type FILE struct {
```
Comment (jpillora) [inline fetcher/fetcher_file.go]: Can you please rename to "File"?

File: fetcher/fetcher_file.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	PATH     string
```
Comment (jpillora) [inline fetcher/fetcher_file.go]: Can you please rename to "Path"?

File: fetcher/fetcher_file.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+	defer file.Close()
+
+	state, err := file.Stat()
+	if err != nil {
+		return fmt.Errorf("Get file state error: %s", err)
+	}
+
+	info := state.Sys().(*syscall.Stat_t)
+	f.uniqTag = fmt.Sprintf("%d%d", info.Ctim.Nano(), info.Ino)
```
Comment (jpillora) [inline fetcher/fetcher_file.go]: To be more OS agnostic (avoid syscall usage), can the tag be made up of: "state.mtime|state.filesize" ? Or if you're keen you could use sha256 :)?

File: fetcher/fetcher_file.go (hunk 1 above)
Comment (wrfly) [inline fetcher/fetcher_file.go]: OK

File: fetcher/fetcher_file.go (hunk 2 above)
Comment (wrfly) [inline fetcher/fetcher_file.go]: Got

File: fetcher/fetcher_file.go (hunk 3 above)
Comment (wrfly) [inline fetcher/fetcher_file.go]: Yeah, I thought of using `sha256` or `md5` before, but they may take a lot of time if the binary is not small, `state.mtime` or `state.filesize` couldn't identify a file sometimes... 

Any suggestions?

I'll make it to `state.mtime|state.filesize` temporarily.

Comment (jpillora) [inline fetcher/fetcher_file.go]: Hashing isn't too bad, SSDs read the disk at GB/sec so I don't think its a big issue. overseer is [already hashing the binary here](https://github.com/jpillora/overseer/blob/master/proc_master.go#L83-L90). Nevertheless, I think it's safe to use `state.mtime|state.filesize`

**Comment by jpillora** [comment]
Thanks!

**Pull request merged**